File: packages/java/tests/gradle/kotlin-gradle-test/src/main/kotlin/com/vaadin/hilla/kotlin/nullable/test/services/ChildEndpoint.kt
package com.vaadin.hilla.kotlin.nullable.test.services

import com.vaadin.hilla.Endpoint

@Endpoint
class ChildEndpoint : ParentEndpoint() {

    fun childMethod1(name: String?, age: Int?): String {
        return if (name?.isBlank() == true) {
            "Hello stranger"
        } else {
            "Hello $name"
        }
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/nonnull/kotlin/extended/snapshots/ChildEndpoint.ts
import { EndpointRequestInit } from "@vaadin/hilla-frontend";
import client from "./connect-client.default.js";
async function childMethod1(name: string | undefined, age: number | undefined, init?: EndpointRequestInit): Promise<string> { return client.call("ChildEndpoint", "childMethod1", { name, age }, init); }
export { childMethod1 };
